parsers/cxx/cxx_parser_tokenizer.c

```diff
@@ -1202,6 +1202,14 @@ void cxxParserUngetCurrentToken(void)
 
 #define CXX_PARSER_MAXIMUM_TOKEN_CHAIN_SIZE 16384
 
+// We stop applying macro replacements if the unget buffer gets too big
+// as it is a sign of recursive macro expansion
+#define CXX_PARSER_MAXIMUM_UNGET_BUFFER_SIZE_FOR_MACRO_REPLACEMENTS 65536
+
+// We stop applying macro replacements if a macro is used so many
+// times in a recursive macro expansion.
+#define CXX_PARSER_MAXIMUM_MACRO_USE_COUNT 8
```

Comment:
I suggest to making this far higher. Like 1024 or something.
It will still pass the test but will allow for more complex macros to be used (like big tables buit by macros).

Reply:
This definition limits per-macro usage in a recursive expansion. It doesn't limit the usage of all macros in a recursive expansion. I think even 8 is large. At the early stage of development, I used 1 for the definition.

See 
https://github.com/universal-ctags/ctags/blob/7f7cb386009aed6738bb6aab3434671db0b16fe8/Units/parser-c.r/recursive-macros-2.d/expected.tags

1024 may make the expected.tags very large.

Do you still think the larger value is better?

Reply:
I didn't change CXX_PARSER_MAXIMUM_MACRO_USE_COUNT in the update.